okhttp/src/main/java/okhttp3/Headers.java

```diff
@@ -365,11 +364,12 @@ Builder addLenient(String name, String value) {
     }
 
     public Builder removeAll(String name) {
-      for (int i = 0; i < namesAndValues.size(); i += 2) {
+      for (int i = 0, size = namesAndValues.size(); i < size; i += 2) {
         if (name.equalsIgnoreCase(namesAndValues.get(i))) {
           namesAndValues.remove(i); // name
           namesAndValues.remove(i); // value
           i -= 2;
+          size -= 2;
```

Comment:
My preference is to just do a normal for loop whenever the size changes in the loop body.